coremltools/converters/keras/_layers2.py

```diff
@@ -75,9 +75,7 @@ def _get_elementwise_name_from_keras_layer(keras_layer):
     elif isinstance(keras_layer, _keras.layers.Concatenate):
         if len(keras_layer.input_shape[0]) == 3 and (keras_layer.axis == 1 or keras_layer.axis == -2):
             return 'SEQUENCE_CONCAT'
-        elif len(keras_layer.input_shape[0]) == 4 and (keras_layer.axis == 3 or keras_layer.axis == -1):
-            return 'CONCAT'
-        elif len(keras_layer.input_shape[0]) == 2 and (keras_layer.axis == 1 or keras_layer.axis == -1):
+        elif keras_layer.axis == len(keras_layer.input_shape[0])-1 or keras_layer.axis == -1:
```

Comment:
I do not think this is a correct change. CoreML only supports shapes un to rank 4, it should error out if the input shape is more than that. 
What use case does this change enable which didn't work before?

Reply:
I have concat layer with input shape 3 and I want concat by axis 2. Before I cannot make it.

Reply:
The correct fix for "input shape 3 and I want concat by axis 2" is to insert permute layers. That is, input shape 3 is internally interpreted as (C,H,W) and axis=2 means the concat is on the 'W' axis. Since this is not supported by default, input should be first permuted to (W,H,C) and then do concat and then permute back. 

So the converter should be translating this concat layer to "permute-concat-permute". 

You want to try putting that fix in this PR? Its similar to how flatten in handled in "convert_flatten" function. 
Otherwise, please open an issue with the Keras network to reproduce it and I'll mark it as a bug and put in the fix.